components/CSDClient/Request.php

```diff
@@ -0,0 +1,120 @@
+<?php
+/**
+ * OpenEyes
+ *
+ * (C) Moorfields Eye Hospital NHS Foundation Trust, 2008-2011
+ * (C) OpenEyes Foundation, 2011-2012
+ * This file is part of OpenEyes.
+ * OpenEyes is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version.
+ * OpenEyes is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+ * You should have received a copy of the GNU General Public License along with OpenEyes in a file titled COPYING. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * @package OpenEyes
+ * @link http://www.openeyes.org.uk
+ * @author OpenEyes <info@openeyes.org.uk>
+ * @copyright Copyright (c) 2008-2011, Moorfields Eye Hospital NHS Foundation Trust
+ * @copyright Copyright (c) 2011-2012, OpenEyes Foundation
+ * @license http://www.gnu.org/licenses/gpl-3.0.html The GNU General Public License V3.0
+ */
+
+namespace OEModule\mehstaffdb\components\CSDClient;
+
+abstract class Request implements RequestInterface
+{
+    /** @var string */
+    protected string $csd_api_key;
+    /** @var string */
+    protected string $csd_api_url;
+
+    /**
+     * Child classes must extend this to be able
+     * to automatically configure the class instance
+     *
+     * @return array
+     */
+    protected function getSettingKeys(): array
+    {
+        return [
+            "csd_api_key",
+            "csd_api_url"
+        ];
+    }
+
+    public function __construct()
+    {
+        foreach ($this->getSettingKeys() as $setting_key) {
+            $setting_value = \Yii::app()->params[$setting_key];
+            if (strlen($setting_value) === 0) {
+                throw new \Exception("$setting_key not set");
+            }
+            $this->$setting_key = $setting_value;
+        }
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getTimeout(): int
+    {
+        return 3;
```

Comment:
This value should be a environment variable, or at least a local variable that can be changed on the back end as it may need to be changed and it would be annoying to need to update the code to do so

Reply:
I created an environment variable for this